### PR TITLE
Add merge param

### PR DESCRIPTION
**Pull request by hardikmashru**
## JIRA Ticket(s) if any

* [MOB-8818](https://iterable.atlassian.net/browse/MOB-8818)

## Description

Added merge params to setUserId and setEmail
Added unit tests for the same

## Test Steps

[MOB-8818]: https://iterable.atlassian.net/browse/MOB-8818?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mprew97** [comment]
Can we resolve the merge conflicts here?

File: src/anonymousUserTracking/anonymousUserEventManager.ts [hunk 1]
```diff
@@ -243,14 +243,17 @@ export class AnonymousUserEventManager {
...
-
-        localStorage.removeItem(SHARED_PREFS_ANON_SESSIONS);
-        localStorage.removeItem(SHARED_PREFS_EVENT_LIST_KEY);
-        localStorage.removeItem(SHARED_PREF_MATCHED_CRITERIAS);
+        this.removeAnonSessionNCriteriaData();
       });
     }
   }
 
+  removeAnonSessionNCriteriaData() {
```
Comment (mprew97) [inline src/anonymousUserTracking/anonymousUserEventManager.ts:251]: typo 
```suggestion
  removeAnonSessionCriteriaData() {
```

File: src/anonymousUserTracking/anonymousUserEventManager.ts [hunk 2]
```diff
@@ -149,7 +149,7 @@ export class AnonymousUserEventManager {
     this.storeEventListToLocalStorage(newDataObject, false);
   }
 
-  private async checkCriteriaCompletion() {
+  private checkCriteriaCompletion() {
```
Comment (mprew97) [inline src/anonymousUserTracking/anonymousUserEventManager.ts:152]: can we type these methods out? would be helpful to know all of their return types.

File: src/authorization/authorization.ts [hunk 3]
```diff
@@ -425,15 +429,58 @@ export function initialize(
     }
   };
 
+  const getMergeDefaultValue = (merge?: boolean) => {
+    const doesAnonUserExist = getAnonUserId() === null;
+    return merge === undefined
```
Comment (mprew97) [inline src/authorization/authorization.ts:434]: it's easier to read non-nested ternaries. Can we update this to use basic `if` statements instead?

File: src/authorization/authorization.ts [hunk 4]
```diff
@@ -425,15 +429,58 @@ export function initialize(
...
+      ? (authIdentifier === null && typeOfAuth === null && doesAnonUserExist) || // Criteria is not yet met (default merge is true)
+        (authIdentifier !== null && typeOfAuth !== null && !doesAnonUserExist) // Criteria is met (Iterable profile created with an autogenerated identity)(default merge is true)
+        ? true
+        : false // Current logged in user is identified (default merge is false)
+      : merge;
+  };
+
+  const handleMerge = async (
+    emailOrUserId: string,
```
Comment (mprew97) [inline src/authorization/authorization.ts:443]: can we make this just take the same params as `tryMergeUser`? Then you won't need to do the conversions on line 447-452,

**Review by mprew97 — CHANGES_REQUESTED** [review]

File: src/authorization/authorization.ts (hunk 4 above)
Comment (mprew97) [reply src/authorization/authorization.ts:450]: sorry meant `mergeUser`, not `tryMergeUser`

Comment (hardikmashru) [reply src/authorization/authorization.ts:450]: Actually we need to handle the case in tryMergeUser where we won't call  anonymousUserMerge.mergeUser at all and return promise with true, hence we need this additional layer.

**Review by mprew97 — APPROVED** [review]

**Pull request merged**